yellowbrick/cluster/elbow.py

```diff
@@ -220,19 +222,19 @@ def __init__(self, model, ax=None, k=10,
 
         # Convert K into a tuple argument if an integer
         if isinstance(k, int):
-            k = (2, k+1)
-
-        # Expand k in to the values we will use, capturing exceptions
-        try:
-            k = tuple(k)
+            self.k_values_ = list(range(2, k+1))
+        elif isinstance(k, tuple) and len(k) == 2 and \
+                all(isinstance(x, (int, np.integer)) for x in k):
             self.k_values_ = list(range(*k))
-        except:
+        elif isinstance(k, collections.Iterable) and \
+                all(isinstance(x, (int, np.integer)) for x in k):
+            self.k_values_ = list(k)
+        else:
             raise YellowbrickValueError((
-                "Specify a range or maximal K value, the value '{}' "
-                "is not a valid argument for K.".format(k)
+                "Specify an iterable of integers, a range, or maximal K value,"
+                " the value '{}' is not a valid argument for K.".format(k)
```

Comment:
- This looks good...
- great use of `all` to make sure only integers are present.
- Thanks for updating the Error messages